src/build/index.ts

```diff
@@ -381,6 +384,24 @@ export default class StreamsBuild {
         }
     }
 
+    private static async checkIfDirty(filePath): Promise<void> {
+        let isFileDirty = false;
+        let dirtyFile = null;
+        const fileName = filePath.substring(filePath.lastIndexOf("/") + 1);
```

Comment:
```suggestion
        const fileName = path.basename(filePath);
```
We can use [`path.basename`](https://nodejs.org/api/path.html#path_path_basename_path_ext) to get the name of the file (e.g., `HelloWorld.spl`).